timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -292,55 +289,55 @@ void __device__ v_nonbonded_unified(
 
     const int index = row_block_idx * 32 + threadIdx.x;
     const unsigned int atom_i_idx = index < NR ? row_idxs[index] : N;
-    // int lambda_offset_i = atom_i_idx < N ? lambda_offset_idxs[atom_i_idx] : 0;
-    // int lambda_plane_i = atom_i_idx < N ? lambda_plane_idxs[atom_i_idx] : 0;
 
     RealType ci_x = atom_i_idx < N ? coords[atom_i_idx * 3 + 0] : 0;
     RealType ci_y = atom_i_idx < N ? coords[atom_i_idx * 3 + 1] : 0;
     RealType ci_z = atom_i_idx < N ? coords[atom_i_idx * 3 + 2] : 0;
-    RealType ci_w = atom_i_idx < N ? coords_w[atom_i_idx] : 0;
 
     unsigned long long gi_x = 0;
     unsigned long long gi_y = 0;
     unsigned long long gi_z = 0;
 
-    int charge_param_idx_i = atom_i_idx * 3 + 0;
-    int lj_param_idx_sig_i = atom_i_idx * 3 + 1;
-    int lj_param_idx_eps_i = atom_i_idx * 3 + 2;
+    int charge_param_idx_i = atom_i_idx * PARAMS_PER_ATOM + PARAM_OFFSET_CHARGE;
+    int lj_param_idx_sig_i = atom_i_idx * PARAMS_PER_ATOM + PARAM_OFFSET_SIG;
+    int lj_param_idx_eps_i = atom_i_idx * PARAMS_PER_ATOM + PARAM_OFFSET_EPS;
+    int w_param_idx_i = atom_i_idx * PARAMS_PER_ATOM + PARAM_OFFSET_W;
```

Comment:
Nit: style -- I do see the reason to replace `* 3` with `* PARAMS_PER_ATOM`, but personally these index expressions seem slightly noisier to skim now, just because there are more characters per line...

Perhaps extract repeated `atom_i_idx * PARAMS_PER_ATOM`?

Reply:
I was also kind of annoyed by this and tried out shorter names for the constants, but nothing stuck. I'll give your suggestion of factoring out the common offset a try.

Reply:
Addressed in ea2a474